store/tikv/oracle/oracles/pd.go

```diff
@@ -231,3 +238,46 @@ func (o *pdOracle) GetLowResolutionTimestampAsync(ctx context.Context, opt *orac
 		err: nil,
 	}
 }
+
+func (o *pdOracle) getStaleTimestamp(ctx context.Context, prevSecond uint64) (uint64, error) {
+	ts, ok := o.getLastTS(oracle.GlobalTxnScope)
+	if !ok {
+		return 0, errors.Errorf("get stale timestamp fail, invalid txnScope = %s", oracle.GlobalTxnScope)
+	}
+	tsArrival, ok := o.getLastArrivalTS()
+	if !ok {
+		return 0, errors.Errorf("get last arrival timestamp, invalid txnScope = %s", oracle.GlobalTxnScope)
+	}
+	arrivalTime := oracle.GetTimeFromTS(tsArrival)
+	physicalTime := oracle.GetTimeFromTS(ts)
+	if uint64(physicalTime.Unix()) <= prevSecond {
+		return 0, errors.Errorf("invalid prevSecond %v", prevSecond)
+	}
+
+	staleTime := physicalTime.Add(-arrivalTime.Sub(time.Now().Add(-time.Duration(prevSecond) * time.Second)))
+
+	return oracle.ComposeTS(oracle.GetPhysical(staleTime), 0), nil
+}
+
+// GetStaleTimestamp generate a TSO which represents for the TSO prevSecond secs ago.
+func (o *pdOracle) GetStaleTimestamp(ctx context.Context, prevSecond uint64) (ts uint64, err error) {
+	ts, err = o.getStaleTimestamp(ctx, prevSecond)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	return ts, nil
+}
+
+func (o *pdOracle) setLastArrivalTS(ts uint64) {
+	if o.lastArrivalTSPointer == nil {
+		o.lastArrivalTSPointer = new(uint64)
+	}
+	atomic.StoreUint64(o.lastArrivalTSPointer, ts)
+}
+
+func (o *pdOracle) getLastArrivalTS() (uint64, bool) {
+	if o.lastArrivalTSPointer == nil {
+		return 0, false
+	}
+	return atomic.LoadUint64(o.lastArrivalTSPointer), true
+}
```

Comment:
```suggestion
func (o *pdOracle) setLastArrivalTS(ts uint64) {
	atomic.StoreUint64(&o.lastArrivalTS, ts)
}

func (o *pdOracle) getLastArrivalTS() (uint64, bool) {

	ts:=atomic.LoadUint64(&o.lastArrivalTS)
	if ts >0 {
	    return ts, true
	}
	return 0, false
}
```

Reply:
Thanks for your review, I agree with you.